warehouse/sessions.py

```diff
@@ -145,6 +146,20 @@ def record_auth_timestamp(self):
         self[self._reauth_timestamp_key] = datetime.datetime.now().timestamp()
         self.changed()
 
+    def record_password_timestamp(self, timestamp):
+        self[self._password_timestamp_key] = timestamp
+        self.changed()
+
+    def password_outdated(self, current_password_timestamp):
+        stored_password_timestamp = self.get(self._password_timestamp_key, None)
+
+        if stored_password_timestamp is None:
+            # This session predates invalidation by password reset... since
+            # we cannot say for sure, let it live its life.
+            return False
+
+        return current_password_timestamp != stored_password_timestamp
```

Comment:
I like that the `!=` comparison also covers our bases against time-travelers.

Reply:
![](https://media.giphy.com/media/103gOGGaHcBq3C/giphy.gif)